beluga/include/beluga/algorithm/amcl_core.hpp

```diff
@@ -162,12 +162,12 @@ class Amcl {
    * \return An optional pair containing the estimated pose and covariance after the update,
    *         or std::nullopt if no update was performed.
    */
-  auto update(state_type control_action, measurement_type measurement) -> std::optional<estimation_type> {
+  auto update(TimeStamped<state_type> control_action, measurement_type measurement) -> std::optional<estimation_type> {
```

Comment:
@fbattocchia rather than forcing this here, take the control action type from the motion model like we do for the measurement type.